src/dodal/beamlines/i23.py

```diff
@@ -27,9 +26,16 @@ def oav_pin_tip_detection(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
 ) -> PinTipDetection:
     """Get the i23 OAV pin-tip detection device"""
+
+    if get_beamline_name("") != "i23":
```

Comment:
Should: This still doesn't really fix the system tests as they will pretend to be i23. I think we should be able to run the system tests from a developer machine but it just skips this device. You should also be able to use the `skip_device` decorator for this